src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -94,16 +126,19 @@ internal async Task SearchAsync(bool searchCurrentDocument, CancellationToken ca
                     // totally hydrated the oop side.  Until that happens, we'll attempt to return cached data from languages
                     // that support that.
                     isFullyLoaded = await _host.IsFullyLoadedAsync(cancellationToken).ConfigureAwait(false);
-                    await SearchAllProjectsAsync(isFullyLoaded, cancellationToken).ConfigureAwait(false);
+                    await SearchAllProjectsAsync(isFullyLoaded, scope, cancellationToken).ConfigureAwait(false);
                 }
             }
             catch (OperationCanceledException)
             {
             }
             finally
             {
-                // providing this extra information will make UI to show indication to users
-                // that result might not contain full data
+                // Ensure that we actually complete all our remaining progress items so that the progress bar completes.
+                await ProgressItemsCompletedAsync(_remainingProgressItems, cancellationToken).ConfigureAwait(false);
```

Comment:
so is the only case where _remainingProgressItems would not be zero is when an exception is thrown and we aren't able to complete the search?

Reply:
not sure how possible this is, but could this race?  For example we have 5 searches remaining, one document throws triggering this catch (we subtract 5 from the remaining), but then one of the previously in progress document searches also completes, subtracting another 1 and we get -1.

Reply:
> so is the only case where _remainingProgressItems would not be zero is when an exception is thrown and we aren't able to complete the search?

No.  A totally normal thing that causes this is if we also shortcircuit remaining work.  i can add comment, btu this happens if we search cached documents and find results (during solution load).  in that case, we skip the other searches.